src/main/java/org/opentripplanner/updater/trip/TimetableSnapshotSource.java

```diff
@@ -690,12 +699,92 @@ private Result<?, UpdateError> handleAddedTrip(
     return addTripToGraphAndBuffer(
       tripBuilder.build(),
       tripUpdate,
+      stopTimeUpdates,
       stops,
       serviceDate,
       RealTimeState.ADDED
     );
   }
 
+  private Route getOrCreateRoute(TripDescriptor tripDescriptor, FeedScopedId tripId) {
+    // Create dummy agency for added trips
+    Agency fallbackAgency = Agency
+      .of(new FeedScopedId(tripId.getFeedId(), "autogenerated-gtfs-rt-added-route"))
+      .withName("Agency automatically added by GTFS-RT update")
+      .withTimezone(transitService.getTimeZone().toString())
```

Comment:
All of the agencies in a feed must have the same timezone per the specification, so we should just fetch the first one in the feed.

Also, if the feed contains just one agency, we should probably always use that.